src/WithdrawalQueueManager.sol

```diff
@@ -0,0 +1,353 @@
+        uint256 amount,
+        uint256 redemptionRateAtRequestTime
+    ) public view returns (uint256) {
+        return amount * redemptionRateAtRequestTime / (10 ** redeemableAsset.decimals());
+    }
+
+
+    /// @notice Calculates the withdrawal fee based on the amount and the current fee percentage.
+    /// @param amount The amount from which the fee should be calculated.
+    /// @return fee The calculated fee.
+    function calculateFee(uint256 amount, uint256 requestWithdrawalFee) public view returns (uint256) {
+        return (amount * requestWithdrawalFee) / FEE_PRECISION;
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  REDEMPTION ASSETS  -------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /// @notice Calculates the surplus of redemption assets after accounting for all pending withdrawals.
+    /// @return surplus The amount of surplus redemption assets in the unit of account.
+    function surplusRedemptionAssets() public view returns (uint256) {
+        uint256 availableAmount = redemptionAssetsVault.availableRedemptionAssets();
+        if (availableAmount > pendingRequestedRedemptionAmount) {
+            return availableAmount - pendingRequestedRedemptionAmount;
+        } 
+        
+        return 0;
+    }
+
+    /// @notice Calculates the deficit of redemption assets after accounting for all pending withdrawals.
+    /// @return deficit The amount of deficit redemption assets in the unit of account.
+    function deficitRedemptionAssets() public view returns (uint256) {
+        uint256 availableAmount = redemptionAssetsVault.availableRedemptionAssets();
+        if (pendingRequestedRedemptionAmount > availableAmount) {
+            return pendingRequestedRedemptionAmount - availableAmount;
+        }
+        
+        return 0;
+    }
+
+    /// @notice Withdraws surplus redemption assets to a specified address.
+    function withdrawSurplusRedemptionAssets(uint256 amount) external onlyRole(REDEMPTION_ASSET_WITHDRAWER_ROLE) {
+        uint256 surplus = surplusRedemptionAssets();
+        if (amount > surplus) {
+            revert AmountExceedsSurplus(amount, surplus);
+        }
+        redemptionAssetsVault.withdrawRedemptionAssets(amount);
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  FINALITY  ----------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Checks if a withdrawal request with a given index is finalized.
+     * @param index The index of the withdrawal request.
+     * @return True if the request is finalized, false otherwise.
+     */
+    function withdrawalRequestIsFinalized(uint256 index) public view returns (bool) {
```

Comment:
this function will return true for nonexistent withdrawal requests. 

If you pass in a nonexistent index, then `WithdrawalRequest memory request = withdrawalRequests[index];` will return an empty `WithdrawalRequest`. Then `isFinalized` will check `return block.timestamp >= request.creationTimestamp + secondsToFinalization;`, which will become `some-timestamp >= 0 + secondsToFianlization`. And since `secondsToFinalization` is very very likely to be less than `block.timestamp`, the `isFinalized` check will return true, even though this withdrawal request does not exist.

fix: `if (request.creationTimestamp == 0) revert WithdrawalRequestDoesNotExist(index);`